Assets/BossRoom/Scripts/Client/Game/Character/ClientInputSender.cs

```diff
@@ -169,12 +177,49 @@ void FixedUpdate()
                 {
                     m_LastSentMove = Time.time;
                     var ray = m_MainCamera.ScreenPointToRay(Input.mousePosition);
-                    if (Physics.RaycastNonAlloc(ray, k_CachedHit, k_MouseInputRaycastDistance, k_GroundLayerMask) > 0)
+
+                    // clear previous results
+                    for (int i = 0; i < k_CachedHit.Length; i++)
```

Comment:
why this step? Why not just rely on the count returned by RaycastNonAlloc?

Reply:
Is the array always guaranteed to be populated orderly? I was wanting to access elements within the array, but that could simplify the iteration.

However, I'm revisiting this atm, so this may end up reworked. I think layers will be the cleanest fix (though it will affect prefabs and potentially some gameplay scripts).

Reply:
check the example here https://docs.unity3d.com/ScriptReference/Physics.RaycastNonAlloc.html
you want to use non-alloc for perf reasons, it wouldn't be super performant to have to access all elements to check if they are null :P

Reply:
It's since been reworked and optimized.